operator/internal/manifests/storage/configure.go

```diff
@@ -186,29 +186,58 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
 			},
 		}
 	case lokiv1.ObjectStorageSecretS3:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAWSAccessKeyID,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
+		sts := opts.S3 != nil && len(opts.S3.RoleArn) != 0
+		if !sts {
+			storeEnvVars = []corev1.EnvVar{
+				{
+					Name: EnvAWSAccessKeyID,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSAccessKeyID,
 						},
-						Key: KeyAWSAccessKeyID,
 					},
 				},
-			},
-			{
-				Name: EnvAWSAccessKeySecret,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
+				{
+					Name: EnvAWSAccessKeySecret,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSAccessKeySecret,
 						},
-						Key: KeyAWSAccessKeySecret,
 					},
 				},
-			},
+			}
```

Comment:
Maybe it's time and should have been in done in the first place to separate these monster switch branch contents in small function in the `var.go` file, e.g. `func S3EnvVars(secretName string, isSTS bool){}`. WDYT? If yes we should create a prerequisite PR to backport it.

Reply:
Let me know if it looks more easy to read. I tend to like deep functions with simple interfaces, functions should also be easy to read so if you still feel this too much we can split it